src/plone/restapi/tests/http-examples/content_get_folder.resp

```diff
@@ -22,7 +22,7 @@ Content-Type: application/json
       "@id": "http://localhost:55001/plone/folder/@workflow"
     }
   }, 
-  "@id": "http://localhost:55001/plone/folder?metadata_fields=UID&metadata_fields=Creator", 
```

Comment:
@plone/volto-team I can see that `@id` was initially thought for keeping the request how this response was created... and maintains the "browserability" promised by (is it hydra or jsonld?)
However, the fact that we've been living without it all this time... makes me wonder if maintaining this REST "purity" is no longer relevant.

Reply:
I think the intention of having the `@id` reflect the uniqueness of the conditions that generated that response is very good. But I think this intention is not reflected in the whole stack, because I argue: if the `@id` is important here and it makes things "browserable", then the `@id` everywhere should reflect this. Let me point to this bit of code, which is the default DX content serializer:
```
        result = {
            # '@context': 'http://www.w3.org/ns/hydra/context.jsonld',
            "@id": obj.absolute_url(),
            "id": obj.id,
            "@type": obj.portal_type,
            "parent": parent_summary,
            "created": json_compatible(obj.created()),
            "modified": json_compatible(obj.modified()),
            "review_state": self._get_workflow_state(obj),
            "UID": obj.UID(),
            "version": version,
            "layout": self.context.getLayout(),
            "is_folderish": False,
        }

        # Insert next/prev information
        nextprevious = NextPrevious(obj)
        result.update(
            {"previous_item": nextprevious.previous, "next_item": nextprevious.next}
        )

        # Insert expandable elements
        result.update(expandable_elements(self.context, self.request))
```

Notice that the `result['@id']` is the object url, but that results depends on the expands. Without the expand you can't reproduce the resource identified by that `@id`.

Reply:
The @id is the node identifier from the JSON-LD (https://www.w3.org/TR/json-ld/#node-identifiers) standard and should reference the resource (without attributes as far as I understand it). We could use the absolute_url attr or come up with a new one instead. I don't see a value in just dropping it and replacing it with something non-standard. Just because it is not used everywhere, does not mean it has no value IMHO. It uniquely identifies the endpoint and indeed allows the human user of the API to browse and discover the API.

Reply:
@tisto the `@id` wouldn't be dropped, it's already inserted in the standard serializer. 

My point was that, if the `@id` here is important and should uniquely identify the Hyperbatch, then the same logic should apply to all nodes that represent themselves with that `@id`, which I don't think is true, though, as we have ids attached to resources with varying amount of information, depending on how that resource is fetched (see my expand example in the previous comment). Basically, we have to accept that we're always dealing with partial graphs when fetching the endpoints and the resources with those ids `@id` are incomplete.

Reply:
@tiberiuichim I discussed this with @sneridagh 1:1 in a call and figured that out :)